ERCS/erc-7629.md

```diff
@@ -0,0 +1,296 @@
+
+    /**
+     * @dev Returns the total supply of ERC-20 tokens.
+     */
+    function erc20TotalSupply() external view returns (uint256);
+
+    /**
+     * @dev Returns the balance of an address for ERC-20 tokens.
+     * @param owner The address to query the balance of.
+     */
+    function erc20BalanceOf(address owner) external view returns (uint256);
+
+    /**
+     * @dev Returns the total supply of ERC-721 tokens.
+     */
+    function erc721TotalSupply() external view returns (uint256);
+
+    /**
+     * @dev Returns the balance of an address for ERC-721 tokens.
+     * @param owner The address to query the balance of.
+     */
+    function erc721BalanceOf(address owner) external view returns (uint256);
+
+    /**
+     * @dev Checks if an operator is approved for all tokens of a given owner.
+     * @param owner The address of the token owner.
+     * @param operator The address of the operator to check.
+     */
+    function isApprovedForAll(
+        address owner,
+        address operator
+    ) external view returns (bool);
+
+    /**
+     * @dev Returns the remaining number of tokens that spender will be allowed to spend on behalf of owner.
+     * @param owner The address of the token owner.
+     * @param spender The address of the spender.
+     */
+    function allowance(
+        address owner,
+        address spender
+    ) external view returns (uint256);
+
+    /**
+     * @dev Returns the array of ERC-721 token IDs owned by a specific address.
+     * @param owner The address to query the tokens of.
+     */
+    function owned(address owner) external view returns (uint256[] memory);
+
+    /**
+     * @dev Returns the address that owns a specific ERC-721 token.
+     * @param id The token ID.
+     */
+    function ownerOf(uint256 id) external view returns (address erc721Owner);
+
+    /**
+     * @dev Returns the URI for a specific ERC-721 token.
+     * @param id The token ID.
+     */
+    function tokenURI(uint256 id) external view returns (string memory);
+
+    /**
+     * @dev Approve or disapprove the operator to spend or transfer all of the sender's tokens.
+     * @param spender The address of the spender.
+     * @param amountOrId The amount of ERC-20 tokens or ID of ERC-721 tokens.
+     */
+    function approve(
+        address spender,
+        uint256 amountOrId
+    ) external returns (bool);
+
+    /**
+     * @dev Approve the spender to spend the specified amount of ERC-20 tokens.
+     * @param spender The address of the spender.
+     * @param value The amount of ERC-20 tokens to approve.
+     */
+    function erc20Approve(
+        address spender,
+        uint256 value
+    ) external returns (bool);
+
+    /**
+     * @dev Approve the operator to spend the specified ERC-721 token.
+     * @param spender The address of the spender.
+     * @param id The ID of the ERC-721 token to approve.
+     */
+    function erc721Approve(address spender, uint256 id) external;
```

Comment:
```suggestion
    function erc721Approve(address spender, uint256 tokenId) external;
```
To avoid ambitious, all ERC721 related `id` should be consistently labelled as `tokenId`